client/src/components/UserRegisterForm.js

```diff
@@ -0,0 +1,135 @@
+        body: JSON.stringify({
+          email: this.state.email,
+          name: this.state.name,
+          password: this.state.password,
+          confirmationPassword: this.state.confirmationPassword
+        }),
+        headers: {
+          "Content-Type": "application/json"
+        }
+      };
+      fetch("http://localhost:4000/auth/register", userRequest)
+        .then(res => res.json())
+        .then(this.resetForm);
+    }
+  };
+  resetForm = () => {
+    this.setState({
+      name: "",
+      email: "",
+      password: "",
+      confirmationPassword: "",
+      userSignUp: true
+    });
+  };
+  clickLogin = () => {
+    window.location.href = "/login";
+  };
+  render() {
+    let userSignUp;
+    if (this.state.userSignUp) {
+      userSignUp = "Your account has been successfully created.";
+    } else {
+      userSignUp = null;
+    }
+    return (
+      <Grid centered column={16}>
+        <Grid.Column width={6}>
+          <Header as="h1">User Registration</Header>
+          {this.state.userSignUp ? (
+            <div>
+              {" "}
+              <Message
+                success
+                header="Your user registration was successful"
+                content="You may now log-in with the username you have chosen"
+              />
+              <Button onClick={this.clickLogin}>Login</Button>
+            </div>
+          ) : (
+            <Form onSubmit={this.handleSubmit}>
+              <Form.Field>
+                <label>Name</label>
+                <input
+                  onChange={this.handleChange}
+                  value={this.state.name}
+                  placeholder="Name"
+                  name="name"
+                  required
+                />
+                <label>Email</label>
+                <input
+                  onChange={this.handleChange}
+                  value={this.state.email}
+                  placeholder="Email"
+                  name="email"
+                  required
+                />
+              </Form.Field>
+              <Form.Field>
+                <label>Password</label>
+                <input
+                  onChange={this.handleChange}
+                  placeholder="Password"
+                  name="password"
+                  value={this.state.password}
+                  required
+                />
+              </Form.Field>
+              <Form.Field>
+                <label> Password Confirmation</label>
+                <input
+                  value={this.state.confirmationPassword}
+                  onChange={this.handleChange}
+                  placeholder="Password confirmation"
+                  name="confirmationPassword"
+                  required
+                />
+              </Form.Field>
+              <Form.Field />
+
+              <Button primary type="submit">
+                Submit
+              </Button>
+              <Button secondary>
+                <Link to="/">Cancel</Link>{" "}
```

Comment:
`{" "}` is probably not needed here

Reply:
This was not added by me but i think there is a problem with my format, Luke noticed it after merging but advised that i leave as it is till i get your comment, but i will remove it and see if it would not be repeated again